prototype/mobile_endpoint/models.py

```diff
@@ -0,0 +1,191 @@
+from flask.ext.migrate import Migrate
+from flask.ext.sqlalchemy import SQLAlchemy
+from sqlalchemy.dialects.postgresql import UUID, ARRAY, JSONB
+from mobile_endpoint.case.models import CommCareCase, CommCareCaseIndex
+
+from mobile_endpoint.form.models import doc_types
+
+
+db = SQLAlchemy()
+
+migrate = Migrate()
+
+
+class ToFromGeneric(object):
+    def to_generic(self):
+        raise NotImplementedError()
+
+    @classmethod
+    def from_generic(cls, obj_dict, **kwargs):
+        raise NotImplementedError()
+
+
+case_form_link = db.Table('case_form', db.Model.metadata,
+    db.Column('case_id', UUID(), db.ForeignKey('case_data.id'), primary_key=True),
+    db.Column('form_id', UUID(), db.ForeignKey('form_data.id'), primary_key=True)
+)
+
+
+class FormData(db.Model, ToFromGeneric):
+    __tablename__ = 'form_data'
+    id = db.Column(UUID(), primary_key=True)
+    domain = db.Column(db.Text(), nullable=False, index=True)
+    received_on = db.Column(db.DateTime(), nullable=False)
```

Comment:
(and it should probably be on (domain, received_on) rather than just received_on)